packages/core/src/oidc/init.ts

```diff
@@ -114,6 +114,19 @@ export default async function initOidc(app: Koa): Promise<Provider> {
         return refreshTokenTtl ?? defaultRefreshTokenTtl;
       },
     },
+    extraTokenClaims: async (ctx, token) => {
+      // AccessToken type is not exported by default, need to asset token is AccessToken
+      if (token.kind === 'AccessToken') {
+        const { accountId } = token;
+        const { roleNames } = await findUserById(accountId);
+
+        // Add User Roles to the AccessToken claims. Should be removed once we have RBAC implemented.
+        // User Roles should be hidden and  determined by the AccessToken scope only.
+        return {
+          roles: roleNames,
```

Comment:
should we using `roleNames` across the projects?

Reply:
Not sure if it follows common oidc uses. But since eventually we are gonna migrate to rbac, updated, keep a same name for now.